examples/http-rust-router/src/lib.rs

```diff
@@ -0,0 +1,40 @@
+use anyhow::Result;
+use spin_sdk::{
+    http_component,
+    http::{
+        Request, 
+        Response, 
+        Router, 
+        Params,
+    },
+};
+
+/// A Spin HTTP component that internally routes requests.
+#[http_component]
+fn handle_route(req: Request) -> Result<Response> {
+    let mut router = Router::new();
+    router.get("/hello/:planet", api::hello_planet);
+    router.all("/*", api::echo_wildcard);
+    router.handle(req)
+}
+
+mod api {
+    use super::*;
+
+    // /hello/:planet
+    pub fn hello_planet(_req: Request, params: Params) -> Result<Response> {
+        let planet = params.get("planet").expect("PLANET");
```

Comment:
Many other routers try to autoextract params so that there's no need for this runtime checking which shouldn't fail. Was such an approach considered?

Reply:
Definitely considering it! I have been imagining that such an implementation could be layered over what has been provided here in a follow up iteration.  Is this something you are interested in exploring?